hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -210,21 +211,22 @@ protected void handleDeleteDirEvent(OmDirectoryInfo directoryInfo,
   protected boolean flushAndCommitNSToDB(Map<Long, NSSummary> nsSummaryMap) {
     try {
       writeNSSummariesToDB(nsSummaryMap);
-      nsSummaryMap.clear();
     } catch (IOException e) {
       LOG.error("Unable to write Namespace Summary data in Recon DB.", e);
       return false;
+    } finally {
+      nsSummaryMap.clear();
     }
     return true;
   }
 
-  protected boolean checkAndCallFlushToDB(
+/*  protected boolean checkAndCallFlushToDB(
       Map<Long, NSSummary> nsSummaryMap) {
     // if map contains more than entries, flush to DB and clear the map
     if (null != nsSummaryMap && nsSummaryMap.size() >=
         nsSummaryFlushToDBMaxThreshold) {
       return flushAndCommitNSToDB(nsSummaryMap);
     }
     return true;
-  }
+  }*/
```

Comment:
Do we not need this method anymore?

Reply:
Removed commented code. As discussed, we'll not be tracking metrics at event level and processing time of each task, we can already findout using existing metrics "lastUpdatedTimestamp" if we compute the difference between current and last run of task using Grafana/Prometheus.